workshops/models.py

```diff
@@ -263,6 +263,33 @@ def get_by_natural_key(self, username):
         else:
             return super().get_by_natural_key(username)
 
+    def annotate_with_instructor_eligibility(self):
+        def passed(requirement):
+            return Sum(Case(When(trainingprogress__requirement__name=requirement,
+                                 trainingprogress__state='p',
+                                 trainingprogress__discarded=False,
+                                 then=1),
+                            default=0,
+                            output_field=IntegerField()))
+
+        return self.annotate(
+            passed_training=passed('Training'),
+            passed_swc_homework=passed('SWC Homework'),
+            passed_dc_homework=passed('DC Homework'),
+            passed_discussion=passed('Discussion'),
+            passed_swc_demo=passed('SWC Demo'),
+            passed_dc_demo=passed('DC Demo'),
+        ).annotate(
+            swc_eligible=(F('passed_training') *
+                          F('passed_swc_homework') *
+                          F('passed_discussion') *
+                          F('passed_swc_demo')),
+            dc_eligible=(F('passed_training') *
+                         F('passed_dc_homework') *
+                         F('passed_discussion') *
+                         F('passed_dc_demo')),
```

Comment:
There:
- [`Person.get_missing_swc/dc_instructor_requirements`](https://github.com/swcarpentry/amy/blob/develop/workshops/models.py#L436)
- the trainee dashboard ([template](https://github.com/swcarpentry/amy/blob/develop/workshops/templates/workshops/trainee_dashboard.html), [view](https://github.com/swcarpentry/amy/blob/develop/workshops/views.py#L3415))

There may be other places, but I cannot recall any other one now.